app/src/main/java/de/westnordost/streetcomplete/quests/note_discussion/NoteDiscussionForm.java

```diff
@@ -162,11 +229,61 @@ private void onClickOk()
 
 		Bundle answer = new Bundle();
 		answer.putString(TEXT, noteText);
+		answer.putStringArrayList(IMAGE_PATHS, imagePaths);
 		applyImmediateAnswer(answer);
 	}
 
+	private void takePhoto()
+	{
+		Intent takePhotoIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+		if (takePhotoIntent.resolveActivity(getActivity().getPackageManager()) != null) {
+			try {
+				photoFile = createImageFile();
+			} catch (IOException e) {
+			}
+			if (photoFile != null) {
+				if (Build.VERSION.SDK_INT > 21) {
+					takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, getUriForFile(getActivity(), "de.westnordost.streetcomplete.fileprovider", photoFile));
+				} else {
+					takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, Uri.fromFile(photoFile));
+				}
```

Comment:
Can you explain what this FileProvider is for? Even after reading the documentation about it, I don't get why it is used here.

Reply:
The FileProvider is used for getting a content:// URI for the file.
The documentation says:
> Note: We are using ```getUriForFile(Context, String, File)``` which returns a ```content:// URI```. For more recent apps targeting Android 7.0 (API level 24) and higher, passing a ```file:// URI``` across a package boundary causes a ```FileUriExposedException```. Therefore, we now present a more generic way of storing images using a ```FileProvider```.

Reply:
Hmm, so, providing this url to the camera counts as "across package boundary" then, I guess?
Perhaps you could add a comment there with a link to the documentation, just for future reference.

Reply:
Yes I think so... I will add a link with the next commit

Reply:
Done